agent/api/task_routes.py

```
@@ -172,8 +172,42 @@ async def get_recent_failed_tasks(
+
+    @router.delete("/task-suppressions/{rule_id}", status_code=204)
+    async def delete_task_suppression(rule_id: str, session: Session = Depends(get_db)):
+        deleted = SuppressionService(session).delete_rule(rule_id)
+        if not deleted:
+            raise HTTPException(status_code=404, detail="Suppression rule not found")
+        return None
+
+    @router.get("/tasks/failed/suppression-metrics", response_model=TaskSuppressionMetrics)
+    async def get_failed_task_suppression_metrics(
+        hours: Optional[int] = Query(default=None, ge=1, le=168),
+        session: Session = Depends(get_db),
```

Ruff (0.15.12)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)